src/Command/LintCommand.php

```diff
@@ -28,66 +33,24 @@ protected function configure(): void
 
     protected function execute(InputInterface $input, OutputInterface $output): int
     {
-        $start = microtime(true);
         $path = (string)$input->getArgument(self::ARG_PATH);
-        $files = $this->finder->find($path);
 
-        $errorCount = 0;
+        $start = microtime(true);
 
-        foreach ($files as $fileInfo) {
-            $table = new Table($output);
-            $table->setHeaders([
-                'line', 'col', 'severity', 'message',
-            ]);
-            $diagnostics = iterator_to_array(
-                $this->linter->lint(
-                    $fileInfo->path,
-                    file_get_contents($fileInfo->path)
-                )
+        $featureDiagnosticsList = [];
+        foreach ($this->finder->find($path) as $featureFile) {
+            $featureDiagnosticsList[] = new FeatureDiagnostics(
+                $featureFile,
+                iterator_to_array($this->linter->lint($featureFile->path, $featureFile->contents()))
```

Comment:
could simply return the FeatureDiagnostics from here, but possibly having a generator is useful if used in an async context (e.g. language server)